extensions/ql-vscode/src/query-history-info.ts

```diff
@@ -43,3 +44,22 @@ export function getQueryText(item: QueryHistoryInfo): string {
       assertNever(item);
   }
 }
+
+export function buildRepoLabel(item: RemoteQueryHistoryItem | VariantAnalysisHistoryItem): string {
+  if (item.t === 'remote') {
+      // Return the number of repositories queried if available. Otherwise, use the controller repository name.
+      const repositoryCount = item.remoteQuery.repositoryCount;
+  
+      if (repositoryCount) {
+        return pluralize(repositoryCount, 'repository', 'repositories');
+      }
+      return `${item.remoteQuery.controllerRepository.owner}/${item.remoteQuery.controllerRepository.name}`;
+    } else if (item.t === 'variant-analysis') {
+      const totalScannedRepositoryCount = item.variantAnalysis.scannedRepos?.length ?? 0;
```

Comment:
Note: When we first trigger the variant analysis, `scannedRepos` might be undefined so we currently show `0` in this case. It looks slightly weird to display `0/0 repositories`, but this should update as soon as we start monitoring/updating the variant analysis.

![image](https://user-images.githubusercontent.com/42641846/197229697-c9ffbb32-191e-49f1-b7e7-233a96342c0a.png)


------

Additional details, in case people are interested 😅 

Other options we considered for when `scannedRepos` is undefined:
- Controller repo name. The variant analysis only saves the controller repo ID, so we'd have to do some re-jigging to get the name.
- Something generic like `remote repositories`. This could work, but might not be any less confusing than `0/0 repositories`.
- Empty string. Annoyingly, the `on` is hard-coded in the label ([here](https://github.com/github/vscode-codeql/blob/4e5abee2ea2dcd437c0c58c844abca77dfd99be3/extensions/ql-vscode/package.json#L236)), so it would be some effort to adjust this and make the label make sense in all cases.

Other thoughts/suggestions welcome of course!

Reply:
Seems like something we should run past Product/Design or stick with the same approach that we have for remote queries (which I think is your first bullet point). I don't think we should block the PR though as it's taking as a step further.

Reply:
Makes sense! I've left it as-is for now, but happy to make some follow-up changes later 🙇🏽‍♀️